pkg/unit_tests/runtime.go

```diff
@@ -0,0 +1,123 @@
+package unit_tests
+
+import (
+	"net"
+	"os"
+	"path/filepath"
+
+	"github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/k3s/pkg/daemons/control/deps"
+	"github.com/rancher/k3s/pkg/version"
+)
+
+// All certs and other files are stored in /tmp/k3s/<RANDOM_STRING>/
+// for testing
+func GenerateTestDataDir(cnf *config.Control) error {
+	var err error
+	if err = os.MkdirAll(cnf.DataDir, 0700); err != nil {
+		return err
+	}
+	testDir, err := os.MkdirTemp(cnf.DataDir, "*")
+	if err != nil {
+		return err
+	}
+	// Remove old symlink and add new one
+	os.Remove(filepath.Join(cnf.DataDir, "latest"))
+	if err = os.Symlink(testDir, filepath.Join(cnf.DataDir, "latest")); err != nil {
+		return err
+	}
+	cnf.DataDir = testDir
+	cnf.DataDir, err = filepath.Abs(cnf.DataDir)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func CleanupTestDataDir(cnf *config.Control) {
+	os.RemoveAll(cnf.DataDir)
+}
+
+func GenerateTestRuntime(cnf *config.Control) error {
+	var err error
```

Comment:
This can be removed and an colon added on line 54.

Reply:
Fixed